diff --git a/api/audio_codecs/L16/audio_decoder_L16.cc b/api/audio_codecs/L16/audio_decoder_L16.cc
index a03abe26f7..2a746dc514 100644
--- a/api/audio_codecs/L16/audio_decoder_L16.cc
+++ b/api/audio_codecs/L16/audio_decoder_L16.cc
@@ -32,18 +32,14 @@ absl::optional<AudioDecoderL16::Config> AudioDecoderL16::SdpToConfig(
 
 void AudioDecoderL16::AppendSupportedDecoders(
     std::vector<AudioCodecSpec>* specs) {
-  Pcm16BAppendSupportedCodecSpecs(specs);
+  // disabled
 }
 
 std::unique_ptr<AudioDecoder> AudioDecoderL16::MakeAudioDecoder(
     const Config& config,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    return nullptr;
-  }
-  return std::make_unique<AudioDecoderPcm16B>(config.sample_rate_hz,
-                                              config.num_channels);
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/L16/audio_encoder_L16.cc b/api/audio_codecs/L16/audio_encoder_L16.cc
index 20259b9ad8..78d0b0e55d 100644
--- a/api/audio_codecs/L16/audio_encoder_L16.cc
+++ b/api/audio_codecs/L16/audio_encoder_L16.cc
@@ -45,7 +45,7 @@ absl::optional<AudioEncoderL16::Config> AudioEncoderL16::SdpToConfig(
 
 void AudioEncoderL16::AppendSupportedEncoders(
     std::vector<AudioCodecSpec>* specs) {
-  Pcm16BAppendSupportedCodecSpecs(specs);
+  // disabled
 }
 
 AudioCodecInfo AudioEncoderL16::QueryAudioEncoder(
@@ -61,16 +61,7 @@ std::unique_ptr<AudioEncoder> AudioEncoderL16::MakeAudioEncoder(
     int payload_type,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  AudioEncoderPcm16B::Config c;
-  c.sample_rate_hz = config.sample_rate_hz;
-  c.num_channels = config.num_channels;
-  c.frame_size_ms = config.frame_size_ms;
-  c.payload_type = payload_type;
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  return std::make_unique<AudioEncoderPcm16B>(c);
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/builtin_audio_decoder_factory.cc b/api/audio_codecs/builtin_audio_decoder_factory.cc
index 881113d985..db29a840a8 100644
--- a/api/audio_codecs/builtin_audio_decoder_factory.cc
+++ b/api/audio_codecs/builtin_audio_decoder_factory.cc
@@ -13,17 +13,9 @@
 #include <memory>
 #include <vector>
 
-#include "api/audio_codecs/L16/audio_decoder_L16.h"
 #include "api/audio_codecs/audio_decoder_factory_template.h"
-#include "api/audio_codecs/g711/audio_decoder_g711.h"
-#include "api/audio_codecs/g722/audio_decoder_g722.h"
-#if WEBRTC_USE_BUILTIN_ILBC
-#include "api/audio_codecs/ilbc/audio_decoder_ilbc.h"  // nogncheck
-#endif
-#if WEBRTC_USE_BUILTIN_OPUS
 #include "api/audio_codecs/opus/audio_decoder_multi_channel_opus.h"
-#include "api/audio_codecs/opus/audio_decoder_opus.h"  // nogncheck
-#endif
+#include "api/audio_codecs/opus/audio_decoder_opus.h"
 
 namespace webrtc {
 
@@ -51,18 +43,7 @@ struct NotAdvertised {
 
 rtc::scoped_refptr<AudioDecoderFactory> CreateBuiltinAudioDecoderFactory() {
   return CreateAudioDecoderFactory<
-
-#if WEBRTC_USE_BUILTIN_OPUS
-      AudioDecoderOpus, NotAdvertised<AudioDecoderMultiChannelOpus>,
-#endif
-
-      AudioDecoderG722,
-
-#if WEBRTC_USE_BUILTIN_ILBC
-      AudioDecoderIlbc,
-#endif
-
-      AudioDecoderG711, NotAdvertised<AudioDecoderL16>>();
+      AudioDecoderOpus, NotAdvertised<AudioDecoderMultiChannelOpus>>();
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/builtin_audio_encoder_factory.cc b/api/audio_codecs/builtin_audio_encoder_factory.cc
index 4546a2eaee..8f566fd62b 100644
--- a/api/audio_codecs/builtin_audio_encoder_factory.cc
+++ b/api/audio_codecs/builtin_audio_encoder_factory.cc
@@ -13,17 +13,9 @@
 #include <memory>
 #include <vector>
 
-#include "api/audio_codecs/L16/audio_encoder_L16.h"
 #include "api/audio_codecs/audio_encoder_factory_template.h"
-#include "api/audio_codecs/g711/audio_encoder_g711.h"
-#include "api/audio_codecs/g722/audio_encoder_g722.h"
-#if WEBRTC_USE_BUILTIN_ILBC
-#include "api/audio_codecs/ilbc/audio_encoder_ilbc.h"  // nogncheck
-#endif
-#if WEBRTC_USE_BUILTIN_OPUS
 #include "api/audio_codecs/opus/audio_encoder_multi_channel_opus.h"
-#include "api/audio_codecs/opus/audio_encoder_opus.h"  // nogncheck
-#endif
+#include "api/audio_codecs/opus/audio_encoder_opus.h"
 
 namespace webrtc {
 
@@ -57,18 +49,7 @@ struct NotAdvertised {
 
 rtc::scoped_refptr<AudioEncoderFactory> CreateBuiltinAudioEncoderFactory() {
   return CreateAudioEncoderFactory<
-
-#if WEBRTC_USE_BUILTIN_OPUS
-      AudioEncoderOpus, NotAdvertised<AudioEncoderMultiChannelOpus>,
-#endif
-
-      AudioEncoderG722,
-
-#if WEBRTC_USE_BUILTIN_ILBC
-      AudioEncoderIlbc,
-#endif
-
-      AudioEncoderG711, NotAdvertised<AudioEncoderL16>>();
+      AudioEncoderOpus, NotAdvertised<AudioEncoderMultiChannelOpus>>();
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/g711/audio_decoder_g711.cc b/api/audio_codecs/g711/audio_decoder_g711.cc
index 838f7e9624..08a039c908 100644
--- a/api/audio_codecs/g711/audio_decoder_g711.cc
+++ b/api/audio_codecs/g711/audio_decoder_g711.cc
@@ -40,28 +40,14 @@ absl::optional<AudioDecoderG711::Config> AudioDecoderG711::SdpToConfig(
 
 void AudioDecoderG711::AppendSupportedDecoders(
     std::vector<AudioCodecSpec>* specs) {
-  for (const char* type : {"PCMU", "PCMA"}) {
-    specs->push_back({{type, 8000, 1}, {8000, 1, 64000}});
-  }
+  // disabled
 }
 
 std::unique_ptr<AudioDecoder> AudioDecoderG711::MakeAudioDecoder(
     const Config& config,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  switch (config.type) {
-    case Config::Type::kPcmU:
-      return std::make_unique<AudioDecoderPcmU>(config.num_channels);
-    case Config::Type::kPcmA:
-      return std::make_unique<AudioDecoderPcmA>(config.num_channels);
-    default:
-      RTC_DCHECK_NOTREACHED();
-      return nullptr;
-  }
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/g711/audio_encoder_g711.cc b/api/audio_codecs/g711/audio_encoder_g711.cc
index 1dca3b80d3..5165f6a46d 100644
--- a/api/audio_codecs/g711/audio_encoder_g711.cc
+++ b/api/audio_codecs/g711/audio_encoder_g711.cc
@@ -50,9 +50,7 @@ absl::optional<AudioEncoderG711::Config> AudioEncoderG711::SdpToConfig(
 
 void AudioEncoderG711::AppendSupportedEncoders(
     std::vector<AudioCodecSpec>* specs) {
-  for (const char* type : {"PCMU", "PCMA"}) {
-    specs->push_back({{type, 8000, 1}, {8000, 1, 64000}});
-  }
+  // disabled
 }
 
 AudioCodecInfo AudioEncoderG711::QueryAudioEncoder(const Config& config) {
@@ -66,30 +64,7 @@ std::unique_ptr<AudioEncoder> AudioEncoderG711::MakeAudioEncoder(
     int payload_type,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  switch (config.type) {
-    case Config::Type::kPcmU: {
-      AudioEncoderPcmU::Config impl_config;
-      impl_config.num_channels = config.num_channels;
-      impl_config.frame_size_ms = config.frame_size_ms;
-      impl_config.payload_type = payload_type;
-      return std::make_unique<AudioEncoderPcmU>(impl_config);
-    }
-    case Config::Type::kPcmA: {
-      AudioEncoderPcmA::Config impl_config;
-      impl_config.num_channels = config.num_channels;
-      impl_config.frame_size_ms = config.frame_size_ms;
-      impl_config.payload_type = payload_type;
-      return std::make_unique<AudioEncoderPcmA>(impl_config);
-    }
-    default: {
-      RTC_DCHECK_NOTREACHED();
-      return nullptr;
-    }
-  }
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/g722/audio_decoder_g722.cc b/api/audio_codecs/g722/audio_decoder_g722.cc
index ed7163471a..0aa6d9f6cf 100644
--- a/api/audio_codecs/g722/audio_decoder_g722.cc
+++ b/api/audio_codecs/g722/audio_decoder_g722.cc
@@ -31,26 +31,14 @@ absl::optional<AudioDecoderG722::Config> AudioDecoderG722::SdpToConfig(
 
 void AudioDecoderG722::AppendSupportedDecoders(
     std::vector<AudioCodecSpec>* specs) {
-  specs->push_back({{"G722", 8000, 1}, {16000, 1, 64000}});
+  // disabled
 }
 
 std::unique_ptr<AudioDecoder> AudioDecoderG722::MakeAudioDecoder(
     Config config,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  switch (config.num_channels) {
-    case 1:
-      return std::make_unique<AudioDecoderG722Impl>();
-    case 2:
-      return std::make_unique<AudioDecoderG722StereoImpl>();
-    default:
-      RTC_DCHECK_NOTREACHED();
-      return nullptr;
-  }
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/g722/audio_encoder_g722.cc b/api/audio_codecs/g722/audio_encoder_g722.cc
index 56a6c4da6a..4a6dc98761 100644
--- a/api/audio_codecs/g722/audio_encoder_g722.cc
+++ b/api/audio_codecs/g722/audio_encoder_g722.cc
@@ -47,9 +47,7 @@ absl::optional<AudioEncoderG722Config> AudioEncoderG722::SdpToConfig(
 
 void AudioEncoderG722::AppendSupportedEncoders(
     std::vector<AudioCodecSpec>* specs) {
-  const SdpAudioFormat fmt = {"G722", 8000, 1};
-  const AudioCodecInfo info = QueryAudioEncoder(*SdpToConfig(fmt));
-  specs->push_back({fmt, info});
+  // disabled
 }
 
 AudioCodecInfo AudioEncoderG722::QueryAudioEncoder(
@@ -64,11 +62,7 @@ std::unique_ptr<AudioEncoder> AudioEncoderG722::MakeAudioEncoder(
     int payload_type,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  return std::make_unique<AudioEncoderG722Impl>(config, payload_type);
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/ilbc/audio_decoder_ilbc.cc b/api/audio_codecs/ilbc/audio_decoder_ilbc.cc
index c58316903a..2169f9e7dd 100644
--- a/api/audio_codecs/ilbc/audio_decoder_ilbc.cc
+++ b/api/audio_codecs/ilbc/audio_decoder_ilbc.cc
@@ -29,14 +29,14 @@ absl::optional<AudioDecoderIlbc::Config> AudioDecoderIlbc::SdpToConfig(
 
 void AudioDecoderIlbc::AppendSupportedDecoders(
     std::vector<AudioCodecSpec>* specs) {
-  specs->push_back({{"ILBC", 8000, 1}, {8000, 1, 13300}});
+  // disabled
 }
 
 std::unique_ptr<AudioDecoder> AudioDecoderIlbc::MakeAudioDecoder(
     Config config,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  return std::make_unique<AudioDecoderIlbcImpl>();
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
diff --git a/api/audio_codecs/ilbc/audio_encoder_ilbc.cc b/api/audio_codecs/ilbc/audio_encoder_ilbc.cc
index b497948491..156a4b00a8 100644
--- a/api/audio_codecs/ilbc/audio_encoder_ilbc.cc
+++ b/api/audio_codecs/ilbc/audio_encoder_ilbc.cc
@@ -62,9 +62,7 @@ absl::optional<AudioEncoderIlbcConfig> AudioEncoderIlbc::SdpToConfig(
 
 void AudioEncoderIlbc::AppendSupportedEncoders(
     std::vector<AudioCodecSpec>* specs) {
-  const SdpAudioFormat fmt = {"ILBC", 8000, 1};
-  const AudioCodecInfo info = QueryAudioEncoder(*SdpToConfig(fmt));
-  specs->push_back({fmt, info});
+  // disabled
 }
 
 AudioCodecInfo AudioEncoderIlbc::QueryAudioEncoder(
@@ -78,11 +76,7 @@ std::unique_ptr<AudioEncoder> AudioEncoderIlbc::MakeAudioEncoder(
     int payload_type,
     absl::optional<AudioCodecPairId> /*codec_pair_id*/,
     const FieldTrialsView* field_trials) {
-  if (!config.IsOk()) {
-    RTC_DCHECK_NOTREACHED();
-    return nullptr;
-  }
-  return std::make_unique<AudioEncoderIlbcImpl>(config, payload_type);
+  return nullptr;  // disabled
 }
 
 }  // namespace webrtc
