diff --git a/modules/rtp_rtcp/source/rtp_sender_video.cc b/modules/rtp_rtcp/source/rtp_sender_video.cc
index c863db4ccf..b3e5c36c78 100644
--- a/modules/rtp_rtcp/source/rtp_sender_video.cc
+++ b/modules/rtp_rtcp/source/rtp_sender_video.cc
@@ -338,10 +338,7 @@ void RTPSenderVideo::AddRtpHeaderExtensions(const RTPVideoHeader& video_header,
   // value sent.
   // Set rotation when key frame or when changed (to follow standard).
   // Or when different from 0 (to follow current receiver implementation).
-  bool set_video_rotation =
-      video_header.frame_type == VideoFrameType::kVideoFrameKey ||
-      video_header.rotation != last_rotation_ ||
-      video_header.rotation != kVideoRotation_0;
+  bool set_video_rotation = true;
   if (last_packet && set_video_rotation)
     packet->SetExtension<VideoOrientation>(video_header.rotation);
 
