diff --git a/rtc_base/openssl_stream_adapter.cc b/rtc_base/openssl_stream_adapter.cc
index a40c4ec6dc..bb22ef2883 100644
--- a/rtc_base/openssl_stream_adapter.cc
+++ b/rtc_base/openssl_stream_adapter.cc
@@ -105,11 +105,7 @@ Mutex& GlobalClockMutex() {
 }
 #endif
 
-#ifdef DTLS1_3_VERSION
 #define MAX_SSL_PROTOCOL_DTLS SSL_PROTOCOL_DTLS_13
-#else
-#define MAX_SSL_PROTOCOL_DTLS SSL_PROTOCOL_DTLS_12
-#endif
 
 uint16_t GetMaxVersion(SSLMode ssl_mode, SSLProtocolVersion version) {
   switch (ssl_mode) {
@@ -122,11 +118,7 @@ uint16_t GetMaxVersion(SSLMode ssl_mode, SSLProtocolVersion version) {
         case SSL_PROTOCOL_TLS_12:
           return TLS1_2_VERSION;
         case SSL_PROTOCOL_TLS_13:
-#ifdef TLS1_3_VERSION
           return TLS1_3_VERSION;
-#else
-          return TLS1_2_VERSION;
-#endif
       }
     case SSL_MODE_DTLS:
       switch (version) {
@@ -136,11 +128,7 @@ uint16_t GetMaxVersion(SSLMode ssl_mode, SSLProtocolVersion version) {
           return DTLS1_2_VERSION;
         case SSL_PROTOCOL_NOT_GIVEN:
         case SSL_PROTOCOL_DTLS_13:
-#ifdef DTLS1_3_VERSION
           return DTLS1_3_VERSION;
-#else
-          return DTLS1_2_VERSION;
-#endif
       }
   }
 }
@@ -411,11 +399,9 @@ SSLProtocolVersion OpenSSLStreamAdapter::GetSslVersion() const {
     } else if (ssl_version == DTLS1_2_VERSION) {
       return SSL_PROTOCOL_DTLS_12;
     }
-#ifdef DTLS1_3_VERSION
     if (ssl_version == DTLS1_3_VERSION) {
       return SSL_PROTOCOL_DTLS_13;
     }
-#endif
   } else {
     if (ssl_version == TLS1_VERSION) {
       return SSL_PROTOCOL_TLS_10;
@@ -424,11 +410,9 @@ SSLProtocolVersion OpenSSLStreamAdapter::GetSslVersion() const {
     } else if (ssl_version == TLS1_2_VERSION) {
       return SSL_PROTOCOL_TLS_12;
     }
-#ifdef TLS1_3_VERSION
     if (ssl_version == TLS1_3_VERSION) {
       return SSL_PROTOCOL_TLS_13;
     }
-#endif
   }
 
   return SSL_PROTOCOL_NOT_GIVEN;
@@ -1203,9 +1187,10 @@ SSL_CTX* OpenSSLStreamAdapter::SetupSSLContext() {
   // remove HMAC-SHA256 and HMAC-SHA384 cipher suites, not GCM cipher suites
   // with SHA256 or SHA384 as the handshake hash.
   // This matches the list of SSLClientSocketImpl in Chromium.
-  SSL_CTX_set_cipher_list(
-      ctx,
-      "DEFAULT:!NULL:!aNULL:!SHA256:!SHA384:!aECDH:!AESGCM+AES256:!aPSK:!3DES");
+  SSL_CTX_set_cipher_list(ctx,
+                          "ECDHE-ECDSA-CHACHA20-POLY1305:"
+                          "ECDHE-ECDSA-AES256-GCM-SHA384:"
+                          "ECDHE-ECDSA-AES128-GCM-SHA256");
 
   if (!srtp_ciphers_.empty()) {
     if (SSL_CTX_set_tlsext_use_srtp(ctx, srtp_ciphers_.c_str())) {
@@ -1440,11 +1425,7 @@ void OpenSSLStreamAdapter::EnableTimeCallbackForTesting() {
 }
 
 SSLProtocolVersion OpenSSLStreamAdapter::GetMaxSupportedDTLSProtocolVersion() {
-#if defined(OPENSSL_IS_BORINGSSL) && defined(DTLS1_3_VERSION)
   return SSL_PROTOCOL_DTLS_13;
-#else
-  return SSL_PROTOCOL_DTLS_12;
-#endif
 }
 
 bool OpenSSLStreamAdapter::SetSslGroupIds(const std::vector<uint16_t>& groups) {
diff --git a/rtc_base/ssl_stream_adapter.cc b/rtc_base/ssl_stream_adapter.cc
index 8b6004ac71..3e24ee2770 100644
--- a/rtc_base/ssl_stream_adapter.cc
+++ b/rtc_base/ssl_stream_adapter.cc
@@ -125,25 +125,9 @@ SSLStreamAdapter::GetEphemeralKeyExchangeCipherGroupName(uint16_t group_id) {
 std::set<uint16_t>
 SSLStreamAdapter::GetSupportedEphemeralKeyExchangeCipherGroups() {
   return {
-  // It would be nice if BoringSSL had a function like this!
-#ifdef SSL_GROUP_SECP224R1
-      SSL_GROUP_SECP224R1,
-#endif
-#ifdef SSL_GROUP_SECP256R1
-      SSL_GROUP_SECP256R1,
-#endif
-#ifdef SSL_GROUP_SECP384R1
-      SSL_GROUP_SECP384R1,
-#endif
-#ifdef SSL_GROUP_SECP521R1
-      SSL_GROUP_SECP521R1,
-#endif
-#ifdef SSL_GROUP_X25519
-      SSL_GROUP_X25519,
-#endif
-#ifdef SSL_GROUP_X25519_MLKEM768
-      SSL_GROUP_X25519_MLKEM768,
-#endif
+      // It would be nice if BoringSSL had a function like this!
+      SSL_GROUP_X25519,    SSL_GROUP_X25519_MLKEM768, SSL_GROUP_SECP256R1,
+      SSL_GROUP_SECP384R1, SSL_GROUP_SECP521R1,
   };
 }
 
@@ -152,33 +136,12 @@ SSLStreamAdapter::GetDefaultEphemeralKeyExchangeCipherGroups(
     const FieldTrialsView* field_trials) {
   // It would be nice if BoringSSL had a function like this!
   // from boringssl/src/ssl/extensions.cc kDefaultGroups.
-  if (field_trials && field_trials->IsEnabled("WebRTC-EnableDtlsPqc")) {
     return {
-#ifdef SSL_GROUP_X25519_MLKEM768
         SSL_GROUP_X25519_MLKEM768,
-#endif
-#ifdef SSL_GROUP_X25519
         SSL_GROUP_X25519,
-#endif
-#ifdef SSL_GROUP_SECP256R1
         SSL_GROUP_SECP256R1,
-#endif
-#ifdef SSL_GROUP_SECP384R1
         SSL_GROUP_SECP384R1,
-#endif
     };
-  }
-  return {
-#ifdef SSL_GROUP_X25519
-      SSL_GROUP_X25519,
-#endif
-#ifdef SSL_GROUP_SECP256R1
-      SSL_GROUP_SECP256R1,
-#endif
-#ifdef SSL_GROUP_SECP384R1
-      SSL_GROUP_SECP384R1,
-#endif
-  };
 }
 
 // Default shim for backward compat.
