diff --git a/modules/audio_coding/codecs/opus/audio_encoder_opus.cc b/modules/audio_coding/codecs/opus/audio_encoder_opus.cc
index 83f356091c..6db1a44910 100644
--- a/modules/audio_coding/codecs/opus/audio_encoder_opus.cc
+++ b/modules/audio_coding/codecs/opus/audio_encoder_opus.cc
@@ -208,10 +208,11 @@ int GetMultipliedBitrate(int bitrate, const std::vector<float>& multipliers) {
 
 void AudioEncoderOpusImpl::AppendSupportedEncoders(
     std::vector<AudioCodecSpec>* specs) {
-  const SdpAudioFormat fmt = {"opus",
-                              kRtpTimestampRateHz,
-                              2,
-                              {{"minptime", "10"}, {"useinbandfec", "1"}}};
+  const SdpAudioFormat fmt = {
+      "opus",
+      kRtpTimestampRateHz,
+      2,
+      {{"minptime", "10"}, {"useinbandfec", "1"}, {"cbr", "1"}}};
   const AudioCodecInfo info = QueryAudioEncoder(*SdpToConfig(fmt));
   specs->push_back({fmt, info});
 }
