diff --git a/p2p/base/p2p_transport_channel.cc b/p2p/base/p2p_transport_channel.cc
index af4b800930..c07b41790a 100644
--- a/p2p/base/p2p_transport_channel.cc
+++ b/p2p/base/p2p_transport_channel.cc
@@ -16,6 +16,7 @@
 #include <algorithm>
 #include <functional>
 #include <memory>
+#include <regex>
 #include <set>
 #include <utility>
 
@@ -1210,13 +1211,23 @@ void P2PTransportChannel::ResolveHostnameCandidate(const Candidate& candidate) {
     return;
   }
 
+  // Only allow resolving of mDNS hostnames with <UUID4>.local to avoid
+  // any leaks.
+  // See: https://github.com/rtcweb-wg/mdns-ice-candidates/issues/121
+  auto hostname = candidate.address().HostAsSensitiveURIString();
+  if (!std::regex_match(hostname, uuid_local_hostname_regex_)) {
+    RTC_LOG(LS_WARNING) << "Dropping non-UUID ICE candidate hostname "
+                        << hostname;
+    return;
+  }
+
   auto resolver = async_dns_resolver_factory_->Create();
   auto resptr = resolver.get();
   resolvers_.emplace_back(candidate, std::move(resolver));
   resptr->Start(candidate.address(),
                 [this, resptr]() { OnCandidateResolved(resptr); });
   RTC_LOG(LS_INFO) << "Asynchronously resolving ICE candidate hostname "
-                   << candidate.address().HostAsSensitiveURIString();
+                   << hostname;
 }
 
 void P2PTransportChannel::AddRemoteCandidate(const Candidate& candidate) {
diff --git a/p2p/base/p2p_transport_channel.h b/p2p/base/p2p_transport_channel.h
index a0729c163d..96d3670f34 100644
--- a/p2p/base/p2p_transport_channel.h
+++ b/p2p/base/p2p_transport_channel.h
@@ -26,6 +26,7 @@
 #include <algorithm>
 #include <map>
 #include <memory>
+#include <regex>
 #include <set>
 #include <string>
 #include <utility>
@@ -488,6 +489,15 @@ class RTC_EXPORT P2PTransportChannel : public IceTransportInternal,
     std::unique_ptr<webrtc::AsyncDnsResolverInterface> resolver_;
   };
   std::vector<CandidateAndResolver> resolvers_ RTC_GUARDED_BY(network_thread_);
+  std::regex uuid_local_hostname_regex_ RTC_GUARDED_BY(network_thread_) =
+      std::regex{
+          "^"
+          "[a-fA-F0-9]{8}-"
+          "[a-fA-F0-9]{4}-"
+          "[a-fA-F0-9]{4}-"
+          "[a-fA-F0-9]{4}-"
+          "[a-fA-F0-9]{12}\\.local"
+          "$"};
   void FinishAddingRemoteCandidate(const Candidate& new_remote_candidate);
   void OnCandidateResolved(webrtc::AsyncDnsResolverInterface* resolver);
   void AddRemoteCandidateWithResult(
