diff --git a/p2p/base/p2p_transport_channel.cc b/p2p/base/p2p_transport_channel.cc
index 0a0d8c8b1d..76b20573f0 100644
--- a/p2p/base/p2p_transport_channel.cc
+++ b/p2p/base/p2p_transport_channel.cc
@@ -18,6 +18,7 @@
 #include <memory>
 #include <set>
 #include <utility>
+#include <regex>
 
 #include "absl/algorithm/container.h"
 #include "absl/memory/memory.h"
@@ -1255,13 +1256,23 @@ void P2PTransportChannel::ResolveHostnameCandidate(const Candidate& candidate) {
     return;
   }
 
+  // Only allow resolving of mDNS hostnames with <UUID4>.local to avoid
+  // any leaks.
+  // See: https://github.com/rtcweb-wg/mdns-ice-candidates/issues/121
+  auto hostname = candidate.address().HostAsSensitiveURIString();
+  if (!std::regex_match(hostname, uuid_local_hostname_regex_)) {
+    RTC_LOG(LS_WARNING) << "Dropping non-UUID ICE candidate hostname "
+                        << hostname;
+    return;
+  }
+
   auto resolver = async_dns_resolver_factory_->Create();
   auto resptr = resolver.get();
   resolvers_.emplace_back(candidate, std::move(resolver));
   resptr->Start(candidate.address(),
                 [this, resptr]() { OnCandidateResolved(resptr); });
   RTC_LOG(LS_INFO) << "Asynchronously resolving ICE candidate hostname "
-                   << candidate.address().HostAsSensitiveURIString();
+                   << hostname;
 }
 
 void P2PTransportChannel::AddRemoteCandidate(const Candidate& candidate) {
diff --git a/p2p/base/p2p_transport_channel.h b/p2p/base/p2p_transport_channel.h
index d7bcd239df..6342d93a5c 100644
--- a/p2p/base/p2p_transport_channel.h
+++ b/p2p/base/p2p_transport_channel.h
@@ -30,6 +30,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <regex>
 
 #include "absl/base/attributes.h"
 #include "absl/strings/string_view.h"
@@ -561,6 +562,15 @@ class RTC_EXPORT P2PTransportChannel : public IceTransportInternal,
     std::unique_ptr<webrtc::AsyncDnsResolverInterface> resolver_;
   };
   std::vector<CandidateAndResolver> resolvers_ RTC_GUARDED_BY(network_thread_);
+  std::regex uuid_local_hostname_regex_ RTC_GUARDED_BY(network_thread_) = std::regex {
+    "^"
+    "[a-fA-F0-9]{8}-"
+    "[a-fA-F0-9]{4}-"
+    "[a-fA-F0-9]{4}-"
+    "[a-fA-F0-9]{4}-"
+    "[a-fA-F0-9]{12}\\.local"
+    "$"
+  };
   void FinishAddingRemoteCandidate(const Candidate& new_remote_candidate);
   void OnCandidateResolved(webrtc::AsyncDnsResolverInterface* resolver);
   void AddRemoteCandidateWithResult(
