diff --git a/sdk/objc/api/peerconnection/RTCStatisticsReport.h b/sdk/objc/api/peerconnection/RTCStatisticsReport.h
index 6fbd59b112..bdd16bc792 100644
--- a/sdk/objc/api/peerconnection/RTCStatisticsReport.h
+++ b/sdk/objc/api/peerconnection/RTCStatisticsReport.h
@@ -9,12 +9,14 @@
  */
 
 #import <Foundation/Foundation.h>
+#import "RTCMacros.h"
 
 @class RTCStatistics;
 
 NS_ASSUME_NONNULL_BEGIN
 
 /** A statistics report. Encapsulates a number of RTCStatistics objects. */
+RTC_OBJC_EXPORT
 @interface RTCStatisticsReport : NSObject
 
 /** The timestamp of the report in microseconds since 1970-01-01T00:00:00Z. */
@@ -28,6 +30,7 @@ NS_ASSUME_NONNULL_BEGIN
 @end
 
 /** A part of a report (a subreport) covering a certain area. */
+RTC_OBJC_EXPORT
 @interface RTCStatistics : NSObject
 
 /** The id of this subreport, e.g. "RTCMediaStreamTrack_receiver_2". */
